[#180] Provide ledger udev rules with tezos-client

rvem opened the pull request:
## Description
Problem: Users have to manually install udev rules in case they want to
use tezos-client along with the ledger device.

Solution: Install udev rules during post-installation stage of the
tezos-client package.

Note that potentially further changes can be required in case firmware update won't resolve an issue with the
udev rules on the Raspberry Pi.

<!--
Describes the nature of your changes. If they are substantial, you should
further subdivide this into a section describing the problem you are solving and
another describing your solution.
-->

## Related issue(s)

<!--
- Short description of how the PR relates to the issue, including an issue link.
For example
- Fixed #100500 by adding lenses to exported items

Write 'None' if there are no related issues (which is discouraged).
Please use keywords to close related issues if they should be closed:
https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords
-->

Relates #180 (issue will be resolved once updated packages are published)

#### Related changes (conditional)

- [x] I checked whether I should update the [README](../tree/master/README.md)

- [x] I checked whether native packaging works, i.e. native binary packages
  can be successfully built.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

pasqu4le commented on docker/package/packages.py:
```diff
@@ -63,11 +63,34 @@
 
+ledger_udev_postinst = '''
+if [ ! -f /etc/udev/rules.d/20-hw1.rules ]; then
+    cat <<EOF > /etc/udev/rules.d/20-hw1.rules
```
One question to clean this up a bit: would it not be possible to move this into a separate file to use here and not keep all it content in `packages.py`?

Asking as it is static content and this isn't easy to look at :slightly_smiling_face:

itkach replied:
I agree, separate file with rules included as part of the .deb would be nice. I would go even further and suggest that there's value in making a separate .deb package for udev rules so that all tezos packages that have Ledger Nano S functionality could depend on it: client, signer, baker, endorser and accuser. Otherwise baker, for example, would only work with ledger nano s if tezos-client is also installed.

rvem commented:
So, I've updated the script that adds udev rules, so that it's supposed to work on both existing Ubuntu versions and Raspberry Pi OS. Also, udev rules are now added for all packages post-installs that can use ledger (specifically: `tezos-client`, `tezos-signer`, `tezos-baker`, `tezos-endorser`)

pasqu4le commented on docker/package/scripts/udev-rules:
```diff
@@ -0,0 +1,21 @@
+# SPDX-FileCopyrightText: 2018-2021 Ledger <https://www.ledger.com>
+#
+# SPDX-License-Identifier: LicenseRef-Apache-Ledger
+
```
I think it would be worth putting in a comment here the repository and commit where this came from, for future reference.

Also I noticed that compared to a previous diff you removed the 
```
if [ ! -f /etc/udev/rules.d/20-hw1.rules ]; then
```
initial check. Did you find it to be unnecessary?

Overriding the udev config file is not an issue, but perhaps we'd want to avoid making the two `udevadm` calls if possible?

rvem replied:
I've decided to remove `if [ ! -f /etc/udev/rules.d/20-hw1.rules ]; then` in case we'll have to update provided udev rules in future.

rvem replied:
> we'd want to avoid making the two `udevadm` calls if possible

AFAIU, `udevadm` has to be called in case rules were updated, we can avoid calling it by looking at the diff between existing rules and rules that we're going to write, but the overhead for calling `udevadm` doesn't look significant for me

pasqu4le replied:
:ok_hand: sounds reasonable to me

pasqu4le replied:
Oh but I still think this:

> I think it would be worth putting in a comment here the repository and commit where this came from, for future reference.

pasqu4le commented on docker/package/packages.py:
```diff
@@ -237,7 +241,8 @@ def gen_daemon_specific_postrm(daemon_name):
+                                        daemon_postinst_common + gen_daemon_specific_postinst(f"tezos-baker-{proto}") \
+                                            + ledger_udev_postinst,
                                      postrm_steps=gen_daemon_specific_postrm(f"tezos-baker-{proto}")))
     packages.append(OpamBasedPackage(f"tezos-accuser-{proto}", "Daemon for accusing",
```
Is the accuser supposed to have this as well?

rvem replied:
hmm, indeed, missed it somehow, thanks for catching
